src/accelerate/data_loader.py

```diff
@@ -615,15 +616,15 @@ def prepare_data_loader(
             If set to `True`, the datalaoder prepared is only iterated through on the main process and then the batches
             are split and broadcast to each process. Will default to `True` when the underlying dataset is an
             `IterableDataset`, `False` otherwise.
+        even_batches ('bool', defaults to `True`):
+            If set to `True`, in cases where the total batch size across all processes does not exactly divide the
+            dataset, samples at the start of the dataset will be duplicated so the batch can be divided equally among
+            all workers.
 
     Returns:
         `torch.utils.data.dataloader.DataLoader`: A new data loader that will yield the portion of the batches
 
-    <Tip warning={true}>
-
-    This does not support `BatchSampler` with varying batch size yet.
```

Comment:
You should leave the warning but rephrase to state a `BatchSampler` with varying batch size will require `even_batches=False`.

Reply:
Sure, I can update that